docs/querying/dart.md

```diff
@@ -0,0 +1,140 @@
+
+By default, Dart queries include results form published segments and realtime tasks.
+
+## Enable Dart
+
+To enable Dart, add the following line to your `_common/common.runtime.properties` files:
+
+```
+druid.msq.dart.enabled = true
+```
+
+### Configure resource consumption
+
+You can configure the Broker and the Historical to tune Dart's resource consumption. Since Brokers only act as controllers, they don't require substantial resources. Historicals, on the other hand, are processing the queries. More resources for Historicals can result in faster query processing.
+
+For Brokers, you can set the following configs:
+
+| Property name | Description | Default |
+|---|---|---|
+| `druid.msq.dart.controller.concurrentQueries` | Maximum number of query controllers that can run concurrently on that Broker. Additional controllers are queued. Queries can get stuck waiting for each other if the total value on Brokers exceeds the setting on a single Historical (`druid.msq.dart.worker.concurrentQueries` ).| 1 |
+| `druid.msq.dart.query.context.targetPartitionsPerWorker` |Number of available threads on workers (`druid.processing.numThreads`) | 1 (Multithreading is turned off on Historicals) |
+
+
+For Historicals, you can set the following configs:
+
+| Property name | Description | Default Value |
+|---|---|---|
+| `druid.msq.dart.worker.concurrentQueries` | Maximum number of query workers that can run concurrently on that Historical. Set this to a value equal to or larger than `druid.msq.dart.controller.concurrentQueries` on your Brokers. If you don't, queries can get stuck waiting for each other. | Equal to the number of merge buffers |
+| `druid.msq.dart.worker.heapFraction` | Maximum amount of heap available for use across all Dart queries as a decimal. | 0.35 (35% of heap) |
+
+
+## Run a Dart query
+
+Once enabled, you can use Dart in the Druid console or the SQL query API to issue queries.
+
+### Druid console
+
+In the **Query** view, select **Engine: SQL (Dart)** from the engine selector menu.
+
+### API
+
+Dart uses the SQL endpoint `/druid/v2/sql`. To use Dart, include the query context parameter `engine` and set it to `msq-dart`:
+
+<Tabs>
+<TabItem value="SET" label="SET" default>
+
+As part of your query using `SET engine = 'msq-dart'`:
+
+```json
+"query":"SET \"engine\"='msq-dart';\nSELECT\n  user,\n  commentLength,\n  COUNT(*) AS \"COUNT\"\nFROM \"wikipedia\"\nGROUP BY 1, 2\nORDER BY 2 DESC"
+```
+
+</TabItem>
+
+<TabItem value="context_block" label="Context block">
+
+As part of a `context` block: 
+
+```json
+{
+  "query": "SELECT\n  user,\n  commentLength,\n  COUNT(*) AS \"COUNT\"\nFROM \"wikipedia\"\nGROUP BY 1, 2\nORDER BY 2 DESC",
+  "context": {
+    "engine": "msq-dart"
+  }
+}
+```
+
+
+  </TabItem>
+  </Tabs>
+
+## Query context parameters
+
+You can use query context parameters to control Dart's behavior. The following table lists the supported query context parameters:
```

Comment:
Please mention that Dart can also use any other SQL context parameters, except as otherwise noted.